markers/sauce.py

```diff
@@ -18,3 +19,11 @@ def pytest_configure(config):
             config.option.markexpr = 'sauce and ({})'.format(config.option.markexpr)
         else:
             config.option.markexpr = 'sauce'
+
+
+def pytest_runtest_setup(item):
+    browser.ensure_browser_open(item)
```

Comment:
sure, we can pick the browser at setup and pass it to ensure_browser_open() rather passing the item.